### PR TITLE
Add command history design details to DG

### LINKED ISSUE #108: Command History: Update DG
Expected length: 1+ page per person
The description can contain things such as,
* How the feature is implemented (or is going to be implemented).
* Why it is implemented that way.
* Alternatives considered.

See [reference](https://nus-cs2103-ay2021s2.github.io/website/schedule/week10/project.html) for more details.

### PR DESCRIPTION
**Pull request by benedictkhoomw**
## What this does
This PR adds design details and diagrams of the command history subsystem to the DG.

Closes #108 

## How to test
1. `cd` to `docs/` folder of project.
2. Run `bundle exec jekyll serve`.
3. Open the jekyll-hosted website in your browser (default address: http://127.0.0.1:4000).
4. Skim through the _Command History Feature_ subsection of the _Implementation_ section. 
5. See if the explanations and diagrams are clarifying.

### CODE REVIEW

File: docs/diagrams/commandhistory/CmdHistStorageSequenceDiagram.puml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+-[UI_COLOR]> logic : execute("help")
+activate logic LOGIC_COLOR
+
+logic -[LOGIC_COLOR]> logic : parseAndExecute("help")
+activate logic LOGIC_COLOR
+
+logic -[LOGIC_COLOR]-> logic
+deactivate logic
+
+opt command executes successfully
```
Comment (DrWala) [inline docs/diagrams/commandhistory/CmdHistStorageSequenceDiagram.puml:17]: Should this part be reflected via an optional frame, or should the fact that it was successfully executed be an underlying assumption? Maybe the branch here is better explained through an activity diagram, and the sequence diagram should show the "Happy path"? What do you think?

**Review by DrWala — APPROVED** [review]
Besides the 1 comment I've got, everything looks good. We can merge this first, and make changes later when we have more information.

File: docs/diagrams/commandhistory/CmdHistStorageSequenceDiagram.puml (hunk 1 above)
Comment (benedictkhoomw) [reply docs/diagrams/commandhistory/CmdHistStorageSequenceDiagram.puml:17]: Hmm good point. An optional frame might suggest that a command being unsuccessful is the "happy path". My original intention was to convey the fact that unsuccessful commands will not be recorded in the command history, but I guess an activity diagram or a note in the sequence diagram might be more appropriate.

Thank you, I will consider this suggestion seriously for a future update.

**Review by stein414 — APPROVED** [review]
Diagrams were helpful and my brief run through did not show any issues. LGTM

**Pull request merged**